draft-ietf-avtcore-rtp-over-quic.md

```diff
@@ -658,7 +657,7 @@ credits it will have to provide to the sending peer, and how rapidly it must pro
 Senders can also transmit RTP packets in QUIC DATAGRAMs, using
 a QUIC extension described in {{!RFC9221}}.
 DATAGRAMs can only be used if the use of the DATAGRAM extension was successfully negotiated during the QUIC handshake.
-If the QUIC extension was negotiated using a signaling protocol, but that extension was not negotiated during the resulting QUIC handshake, an endpoint can close the connection with the ROQ\_EXPECTATION\_UNMET error code.
+If the QUIC extension was negotiated using a signaling protocol, but that extension was not negotiated during the resulting QUIC handshake, an endpoint MAY close the connection with the ROQ\_EXPECTATION\_UNMET error code.
```

Comment:
Why is this changed to a normative "MAY"? But if we are thinking about normative behavior, why would an endpoint not close the connection immediately? It seems odd to notice a mismatch at signaling time, but not give the peer endpoint any indication that all is not right, until the peer endpoint tries to use the QUIC extension. My understanding of [Section 12.4 in RFC 9000](https://datatracker.ietf.org/doc/html/rfc9000#name-frames-and-frame-types) is that the result would be a connection error of type FRAME_ENCODING_ERROR, and that doesn't point directly to a signaling/negotiation mismatch, which is the real problem.

Reply:
At the end of [section 3.2](https://mengelbart.github.io/rtp-over-quic-draft/draft-ietf-avtcore-rtp-over-quic.html#section-3.2), we say that an application can use streams instead. FRAME_ENCODING_ERROR would only happen if the QUIC stack is sending datagrams even though the datagram extension was not negotiated at connection establishment. In that case, I would expect the QUIC stack to return an error when the application is trying to send datagrams.

Reply:
> At the end of [section 3.2](https://mengelbart.github.io/rtp-over-quic-draft/draft-ietf-avtcore-rtp-over-quic.html#section-3.2), we say that an application can use streams instead. FRAME_ENCODING_ERROR would only happen if the QUIC stack is sending datagrams even though the datagram extension was not negotiated at connection establishment. In that case, I would expect the QUIC stack to return an error when the application is trying to send datagrams.

Oh, right. This is a variation of the same conversation we were having at the Hackathon in San Francisco - sorry for MY confusion. 

One thing that doesn't help me stay unconfused, is that we are talking about a specific extension here, but the text sounds more general. At the end of [Section 3.2](https://mengelbart.github.io/rtp-over-quic-draft/draft-ietf-avtcore-rtp-over-quic.html#name-rtp-with-quic-streams-quic-), we explain this more clearly than we're saying in this text. The point in Section 3.2 is "you have a decision to make. How badly do you need to use DATAGRAMs?" 

```suggestion
If the DATAGRAM extension was negotiated using a signaling protocol, but was not also negotiated during the resulting QUIC handshake, an endpoint can close the connection with the ROQ\_EXPECTATION\_UNMET error code.
```